File: sanghyun/programmers-2/영어 끝말잇기.java
import java.util.Arrays;
import java.util.HashSet;

class Solution {
    public int[] solution(int n, String[] words) {

        int[] answer = new int[2];
        HashSet<String> wordSet = new HashSet<>();
        int order = 1;
        int person = 1;
        wordSet.add(words[0]);
        for (int i = 1; i < words.length; i++) {
            person++;
            // 중복 단어가 나올 경우 중단
            if(wordSet.contains(words[i])){
                return answer = new int[]{person, order};
            }
            wordSet.add(words[i]);

            // 앞 단어의 마지막 알파벳과 일치하지 않으면 중단
            if((words[i-1].charAt(words[i-1].length()-1) != words[i].charAt(0))){
                return answer = new int[]{person, order};
            }
            if(person == n){
                order++;
                person = 0;
            }
            if(i == words.length-1){
                return answer = new int[]{0,0};
            }
        }
        return answer;
    }
}

File: soyeong/programmers-2/영어-끝말잇기.py
def solution(n, words):
    # words는 끝말잇기의 진행상황
    # 1부터 시작하는 이유는 번호가 1부터 시작하여 인덱스 자체를 번호로 사용하기 위함
    for i in range(1, len(words)):
        # 이번 i번째 워드의 첫 알파벳이 이전 번째의 워드의 마지막 알파벳과 다르다면 true
        # or
        # i번째 인덱스 이전에 같은 word가 이미 나왔었다면 true
        if words[i][0] != words[i - 1][-1] or words[i] in words[:i]:
            # 그 순서를 인원수 대로 나눈 +1 탈락하는 사람 번호
            # 그 순서를 인원수 대로 나눈 몫에서 +1 한게 그 사람의 몇번째 차례에서 떨어지는지 숫자
            return [(i % n) + 1, (i // n) + 1]  # 탈락하는 사람 번호, 탈락하는 사람 자신의 몇번째 차례
    else:
        # 아무것도 해당 안되면 바로 탈락!
        return [0, 0]